sdk/storage/Azure.Storage.DataMovement.Blobs/README.md

```diff
@@ -81,17 +64,66 @@ We guarantee that all client instance methods are thread-safe and independent of
 
 ## Examples
 
-Please see the examples for [Blobs][blobs_examples].
+### Examples using BlobContainerClient extension methods to upload and download directories.
+
+Instantiate the BlobContainerClient
+```C# Snippet:ExtensionMethodCreateContainerClient
+BlobServiceClient service = new BlobServiceClient(serviceUri, credential);
+
+BlobContainerClient container = service.GetBlobContainerClient(containerName);
+```
+Upload local directory to the root of the container
+```C# Snippet:ExtensionMethodSimpleUploadToRoot
+DataTransfer transfer = await container.StartUploadDirectoryAsync(localPath);
 
-### Start Upload from Local File to Block Blob
+await transfer.AwaitCompletion();
+```
+Upload local directory to a directory in the container by specifying a directory prefix
+```C# Snippet:ExtensionMethodSimpleUploadToDirectoryPrefix
+DataTransfer transfer2 = await container.StartUploadDirectoryAsync(localPath, blobDirectoryPrefix: Path.GetDirectoryName(localPath));
+
+await transfer2.AwaitCompletion();
+```
+Download the entire container to a local directory
+```C# Snippet:ExtensionMethodSimpleDownloadContainer
+DataTransfer transfer = await container.StartDownloadDirectoryAsync(localDirectoryPath);
+
+await transfer.AwaitCompletion();
+```
+Download a directory in the container by specifying a directory prefix
+```C# Snippet:ExtensionMethodSimpleDownloadContainerDirectory
+DataTransfer tranfer2 = await container.StartDownloadDirectoryAsync(localDirectoryPath2, blobDirectoryPrefix);
+
+await tranfer2.AwaitCompletion();
+```
+
+### Examples using TransferManager upload and download files and directories.
+Create Instance of TransferManager
+```C# Snippet:CreateTransferManagerSimple
+TransferManager transferManager = new TransferManager(new TransferManagerOptions());
+```
+
+Create Instance of TransferManager with Options
+```C# Snippet:CreateTransferManagerWithOptions
+// Create BlobTransferManager with event handler in Options bag
+TransferManagerOptions transferManagerOptions = new TransferManagerOptions();
+ContainerTransferOptions options = new ContainerTransferOptions()
```

Comment:
Just a note that I just merged #35610 which will have some conflicts (be breaking) for this PR now (sorry!). We've combined `ContainerTransferOptions` and `SingleTransferOptions` into a new class `TransferOptions`.